ckanext/datastore/backend/postgres.py

```diff
@@ -2091,16 +2086,17 @@ def search_sql(self, context: Context, data_dict: dict[str, Any]):
         return search_sql(context, data_dict)
 
     def resource_exists(self, id: str) -> bool:
-        resources_sql = sqlalchemy.text(
+        resources_sql = sa.text(
             u'''SELECT 1 FROM "_table_metadata"
             WHERE name = :id AND alias_of IS NULL''')
-        results = self._get_read_engine().execute(resources_sql, id=id)
+        with self._get_read_engine().connect() as conn:
+            results = conn.execute(resources_sql, {"id": id})
         res_exists = results.rowcount > 0
         return res_exists
 
     def resource_id_from_alias(self, alias: str) -> tuple[bool, Optional[str]]:
         real_id: Optional[str] = None
-        resources_sql = sqlalchemy.text(
+        resources_sql = sa.text(
             u'''SELECT alias_of FROM "_table_metadata" WHERE name = :id''')
         results = self._get_read_engine().execute(resources_sql, id=alias)
```

Comment:
it seems we have a mix of write engine and read engine and direct (missing the wrappers).

someparts are in the writing and others are in the reading. (And should not be mixed). since datastore/xloader can split reading and writing ops.